### PR TITLE
[Term Entry] Kotlin Datetime .atDate() #3806

### LINKED ISSUE #3806: [Term Entry] Kotlin Datetime .atDate()
### Reason/inspiration (optional)

We would like a new term entry on the `.atDate()` method for Kotlin: [Datetime](https://www.codecademy.com/resources/docs/kotlin/datetime). The entry should go in a new file under `docs/content/kotlin/concepts/datetime/terms/atDate/atDate.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```kotlin `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by tPayne0647**
### Description

Closes #3806 

This PR introduces a detailed documentation for the .atDate() extension function in Kotlin. The .atDate() method is a crucial utility in Kotlin's date-time API, allowing developers to seamlessly combine a LocalTime instance with a LocalDate instance to create a LocalDateTime object. This documentation aims to provide clear guidance on the usage and application of this method, enhancing the understanding and productivity of developers working with date and time in Kotlin.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/kotlin/concepts/datetime/terms/atDate/atDate.md
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/atDate/atDate.md:16]:
```suggestion
The **`.atDate()`** method in Kotlin is a function that combines date and time into a `LocalDateTime` object. This method is beneficial when creating a complete timestamp by merging separate `LocalDate` and `LocalTime` instances.
```
Avoid using first- and second-person pronouns (e.g. "I", "we", "you") if possible.

File: content/kotlin/concepts/datetime/terms/atDate/atDate.md
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/atDate/atDate.md:42]:
```suggestion
import java.time.LocalDate
import java.time.LocalTime
import java.time.LocalDateTime

fun main() {
  val date = LocalDate.of(2023, 12, 25)
  val time = LocalTime.of(15, 30)
  val dateTime = time.atDate(date)

  println(dateTime)
}
```

File: content/kotlin/concepts/datetime/terms/atDate/atDate.md
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/atDate/atDate.md:45]:
```suggestion
The above example will give the following output:
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Few changes from my side for you @tPayne0647 :)

File: content/kotlin/concepts/datetime/terms/atDate/atDate.md
Comment (tPayne0647) [reply content/kotlin/concepts/datetime/terms/atDate/atDate.md:45]: Thanks for taking the time to review! I will fix those and sync it.

**Review by dakshdeepHERE — APPROVED** [review]
This looks good for second review 🚀

**Comment by tPayne0647** [comment]
Thanks! I'm excited to contribute 😊

**Review by dakshdeepHERE — APPROVED** [review]
This looks good for second review 🚀

**Comment by tPayne0647** [comment]
Are we just waiting on 1 more reviewer to approve the workflows?

**Comment by dakshdeepHERE** [comment]
Yes correct @tPayne0647

**Comment by tPayne0647** [comment]
Looks like check failed because of an extra blank line at the end. Not sure why the other check failed... I deleted the extra blank line, but looks like someone has to approve the workflow again.

**Comment by tPayne0647** [comment]
Just curious, are the workflows supposed to automatically fix linting/ formatting issues? The initial checks failed because of an extra blank line at the ned and it looked like the workflow was trying to run a command to fix it, but failed to run. Or maybe I don't understand the workflow correctly 😅 

But all is good now 🎉🍾

Merry Christmas btw!!

**Comment by dakshdeepHERE** [comment]
Hello @tPayne0647, Thank you for the wishes 😀 and you don't have to worry about the workflow. We maintainers will look into it from our end. You just have to see through the entries. Thank you for your contributions.

**Pull request merged**
